akka-stream/src/main/scala/akka/stream/impl/ContextPropagation.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.impl
+
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] trait ContextPropagation {
+  def suspendContext(): Unit
+  def resumeContext(): Unit
+}
+
+private[akka] object ContextPropagation {
+  def apply(): ContextPropagation = new ContextPropagationImpl
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] final class ContextPropagationImpl extends ContextPropagation {
+  private val buffer = Buffer[Unit](1, 1)
+  def suspendContext(): Unit = {
+    buffer.enqueue(())
```

Comment:
:+1:

Reply:
Hmm, maybe I misunderstood but I thought we'd have a no-op here and cinnamon would sneak in the actual context propagation logic rather than the 1 elem buffer here?